### PR TITLE
Upgrade pandas to 1.1.0 and update .week usage

### PR DESCRIPTION
**Pull request by wsankey**
### Pull Request Description
Addresses resolving the deprecation warning in the primitive cited in this issue: https://github.com/FeatureLabs/featuretools/issues/1091. Tested via ipython (images below) and message is resolved.

### Before
deprecated

### After
nomoremessage

-----
*After creating the pull request: in order to pass the **changelog_updated** check you will need to update the "Future Release" section of* `docs/source/changelog.rst` *to include this pull request.*

### CODE REVIEW

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ Changelog
         * Change default branch to ``main`` (:pr:`1038`)
         * Raise TypeError if improper input is supplied to ``Entity.delete_variables()`` (:pr:`1064`)
         * Updates for compatibility with pandas 1.1.0 (:pr:`1079`, :pr:`1089`)
+        * Upgrade pandas to >=1.1.0 (:pr: `1094`)
```
Comment (thehomebrewnerd) [inline docs/source/changelog.rst:20]: Also, remove the space between `:pr:` and ``1094``

Comment (wsankey) [reply docs/source/changelog.rst]: Ah that's what it was, thank you!

File: featuretools/primitives/standard/transform_primitive.py [hunk 2]
```diff
@@ -387,7 +390,7 @@ class TimeSince(TransformPrimitive):
         [-1000, -1000000000, -120000000000]
     """
     name = 'time_since'
-    input_types = [Datetime]
+    input_types = [[DatetimeTimeIndex], [Datetime]]
```
Comment (rwedge) [inline featuretools/primitives/standard/transform_primitive.py:393]: this change looks unrelated

Comment (wsankey) [reply featuretools/primitives/standard/transform_primitive.py]: Yes, unrelated. I had checked out this file from master, not main. Reverted.

**Comment by wsankey** [comment]
The message persists for me in the ipython terminal with this change. Changing the line to `warnings.filterwarnings("ignore", category=FutureWarning)` resolves it.

**Comment by wsankey** [comment]
nomsg

**Comment by thehomebrewnerd** [comment]
> The message persists for me in the ipython terminal with this change. Changing the line to `warnings.filterwarnings("ignore", category=FutureWarning)` resolves it within the ipython terminal. This may be filtered with that line within the module but I haven't checked in that context.

I think that change will cause any FutureWarning afterward to be suppressed - so if the user did something outside of featuretools that would generate a different FutureWarning, the would never see it. We were trying to limit scope to the specific message within featuretools. I tested from a command line script and it seemed to work, but didn't check in ipython.

**Comment by rwedge** [comment]
@wsankey when you were testing before were you defining the Week primitive in the ipython or importing it from the featuretools module?   It's possible if you defined it in memory it did not get filtered due to the `module="featuretools"` part of the filter

**Comment by wsankey** [comment]
Understood. Yes that was the issue. Setting the specific warning message and importing the class resolved this issue. Thank you for the guidance @thehomebrewnerd and @rwedge!

Updated PR to be limited in scope.

**Comment by rwedge** [comment]
We should probably add a test case to confirm the warning is being filtered out

I believe this tests that a warning does not occur

```python
with pytest.warns(None) as record):
    function()

assert not record
```

@wsankey would you be willing to write a test case?

**Comment by wsankey** [comment]
Yes, sure @rwedge I just added a unit test. Not certain this is the proper spot but I manually checked it failed, then passed.

File: docs/source/changelog.rst [hunk 3]
```diff
@@ -17,6 +17,7 @@ Changelog
         * Change default branch to ``main`` (:pr:`1038`)
         * Raise TypeError if improper input is supplied to ``Entity.delete_variables()`` (:pr:`1064`)
         * Updates for compatibility with pandas 1.1.0 (:pr:`1079`, :pr:`1089`)
+        * Set pandas version to pandas>=0.24.1,<2.0.0. Filter week and weekofyear primitive warning. (:pr:`1094`)
```
Comment (rwedge) [inline docs/source/changelog.rst:20]: Let's make this "Set pandas version to pandas>=0.24.1,<2.0.0. Filter pandas deprecation warning in Week primitive."

File: featuretools/primitives/standard/transform_primitive.py [hunk 4]
```diff
@@ -212,6 +214,7 @@ class Week(TransformPrimitive):
 
     def get_function(self):
         def week(vals):
+            warnings.filterwarnings("ignore", message="Series.dt.weekofyear and Series.dt.week have been deprecated.", module="featuretools")
```
Comment (rwedge) [inline featuretools/primitives/standard/transform_primitive.py:217]: this line is a little long (142), can we reduce line length

File: featuretools/tests/primitive_tests/test_transform_primitive.py [hunk 5]
```diff
@@ -83,3 +83,14 @@ def test_age_nan():
     ages = age(dates, time=datetime(2020, 2, 26))
     correct_ages = [10.159, np.nan, 8.159]
     np.testing.assert_array_almost_equal(ages, correct_ages, decimal=3)
+
+
+def test_no_deprecation_message():
```
Comment (rwedge) [inline featuretools/tests/primitive_tests/test_transform_primitive.py:88]: let's call this "test_week_no_deprecation_message"

Comment (wsankey) [reply featuretools/tests/primitive_tests/test_transform_primitive.py:88]: :+1:

File: featuretools/primitives/standard/transform_primitive.py (hunk 4 above)
Comment (wsankey) [reply featuretools/primitives/standard/transform_primitive.py:217]: :+1:

**Comment by wsankey** [comment]
Thanks! Anyway to rerun these tests, I think it was a coincidence they failed..

**Comment by wsankey** [comment]
OK thank you!

**Comment by rwedge** [comment]
> Thanks! Anyway to rerun these tests, I think it was a coincidence they failed..

Only the code coverage test failed and that was due to a known issue, wouldn't have been impacted by these code changes
